### PR TITLE
[Support new order funnel - Part 2] Miscellaneous fixes for Dedicated

### PR DESCRIPTION
**Pull request by FredericEspiau**
# Support for new order funnel
This PR is part of the **Support new order funnel** project
## Part 2 - Miscellaneous fixes for Dedicated
It looks like there is a lot of changes but every commit contains only a few character changes.
If you remove whitespaces while reviewing, then you will see that not a lot have changed.

### :nail_care: Refactor
* app.routes.js
    * change indentation
    * reorder parameters
* remove unused code
    * `1000v`
* `User` and `Security` tab
    * change indentation
    * reorder parameters
    * turn functions into class

### :bug: Bugfix
* app.routes.js
    * add dependency injection annotation
* `User` and `Security` tab
    * display error messages properly
    * prevent exposing classes globally
    * use promises better by returning them
    * close dialogs when needed
    * remove confusing messages
    * changes about enabling/disabling users
* `User` tab
    * success message works correctly when creating user
    * the action menu displays only if the user was correctly delivered
    * minor text changes

## References
- MANAGER-2367

### CODE REVIEW

File: client/app/dedicatedCloud/user/disable/dedicatedCloud-user-disable.controller.js [hunk 1]
```diff
@@ -1,16 +1,34 @@
 angular
   .module('App')
-  .controller('DedicatedCloudUserDisableCtrl', ($scope, $stateParams, $translate, DedicatedCloud) => {
-    $scope.user = $scope.currentActionData;
-    $scope.disableUser = function () {
-      $scope.resetAction();
-      DedicatedCloud.disableUser($stateParams.productId, $scope.user.userId).then(
+  .controller('DedicatedCloudUserDisableCtrl', class {
```
Comment (marie-j) [inline client/app/dedicatedCloud/user/disable/dedicatedCloud-user-disable.controller.js]: You could've extracted the `class` `export` and name it and declare it as an `angular` dependency at the end of the file (or even modularize it 😄 )

Comment (FredericEspiau) [reply client/app/dedicatedCloud/user/disable/dedicatedCloud-user-disable.controller.js]: But if I do that, @cbourgois won't have to do it, I don't want to spoil him

File: client/app/dedicatedCloud/translations/Messages_fr_FR.xml [hunk 2]
```diff
@@ -461,9 +461,9 @@
...
-   <translation id="dedicatedCloud_configuration_security_max_concurrent_connections_title" qtlid="147165">Nombre de connexions simultanées</translation>
-   <translation id="dedicatedCloud_configuration_security_max_concurrent_connections_step1_question" qtlid="148045">Veuillez saisir le nombre maximum de connexions d’utilisateurs simultanées :</translation>
-   <translation id="dedicatedCloud_configuration_security_max_concurrent_connections_step2_question" qtlid="183567">Vous allez modifier le nombre de connexions d’utilisateurs simultanées de <br/><b>{0}</b> à <b>{1}</b></translation>
+   <translation id="dedicatedCloud_configuration_security_max_concurrent_connections_title">Modifier le nombre de connexions simultanées</translation>
+   <translation id="dedicatedCloud_configuration_security_max_concurrent_connections_step1_question">Nombre maximum de connexions d’utilisateurs simultanées</translation>
+   <translation id="dedicatedCloud_configuration_security_max_concurrent_connections_step2_question">Vous allez modifier le nombre de connexions d’utilisateurs simultanées de <br/><b>{0}</b> à <b>{1}</b>.</translation>
```
Comment (JeremyDec) [inline client/app/dedicatedCloud/translations/Messages_fr_FR.xml]: There is no way to remove these HTML markups ?

File: client/app/dedicatedCloud/dedicatedCloud.controller.js [hunk 3]
```diff
@@ -1,220 +1,233 @@
...
-    '$translate',
-    '$uibModal',
-    'constants',
-    'DedicatedCloud',
-    'OvhApiDedicatedCloud',
-    'step',
-    'User',
-    function (
-      $log,
+  .controller('DedicatedCloudCtrl', class {
```
Comment (JeremyDec) [inline client/app/dedicatedCloud/dedicatedCloud.controller.js]:
```suggestion
  .controller('DedicatedCloudCtrl', /* @ngInject */ class {
```

File: client/app/dedicatedCloud/translations/Messages_fr_FR.xml [hunk 4]
```diff
@@ -441,8 +441,8 @@
    <translation id="dedicatedCloud_SECURITY_change_nb_simultaneous_connection_success" qtlid="146944">Le nombre maximum de connexions simultanées a été modifié avec succès.</translation>
    <translation id="dedicatedCloud_SECURITY_change_nb_simultaneous_connection_failure" qtlid="146957">Une erreur est survenue lors de la modification du nombre maximum de connexions simultanées.</translation>
    <translation id="dedicatedCloud_SECURITY_change_logout_policy">Modifier la politique de déconnexion</translation>
-   <translation id="dedicatedCloud_SECURITY_add_IP" qtlid="146970">Ajout des IP</translation>
-   <translation id="dedicatedCloud_SECURITY_delete_IP" qtlid="146983">Supprimer des IP</translation>
+   <translation id="dedicatedCloud_SECURITY_add_IP">Ajouter une nouvelle plage d'adresse IP</translation>
```
Comment (JeremyDec) [inline client/app/dedicatedCloud/translations/Messages_fr_FR.xml]:
```suggestion
   <translation id="dedicatedCloud_SECURITY_add_IP">Ajouter une nouvelle plage d'adresses IP</translation>
```

File: client/app/dedicatedCloud/security/dedicatedCloud-security.html [hunk 5]
```diff
@@ -81,9 +81,11 @@ <h2 data-translate="dedicatedCloud_tab_security"></h2>
                                 <td data-ng-bind="policy.description"></td>
                                 <td class="text-center">
 
-                                    <div class="btn-group"
-                                         data-uib-dropdown
-                                         data-dropdown-append-to-body>
+                                    <div
+                                        data-ng-if="policy.status === 'ALLOWED'"
```
Comment (JeremyDec) [inline client/app/dedicatedCloud/security/dedicatedCloud-security.html]: constant ?

File: client/app/dedicatedCloud/security/access/dedicatedCloud-security-access.controller.js [hunk 6]
```diff
@@ -1,51 +1,60 @@
...
-    this.$translate = $translate;
+import _ from 'lodash';
 
-    this.selectedAccessPolicy = {
-      policies: [],
-      value: null,
-    };
+{
+  class DedicatedCloudSecurityPolicyAccessCtrl {
+    constructor(
```
Comment (JeremyDec) [inline client/app/dedicatedCloud/security/access/dedicatedCloud-security-access.controller.js]:
```suggestion
    /* @ngInject */
    constructor(
```

File: client/app/dedicatedCloud/security/add/dedicatedCloud-security-add.controller.js [hunk 7]
```diff
@@ -1,18 +1,41 @@
-angular.module('App').controller('DedicatedCloudSecurityPolicyAddCtrl', ($scope, $stateParams, DedicatedCloud, $translate, REGEX) => {
-  $scope.regex = REGEX;
-  $scope.newNetwork = {
-    value: null,
-  };
+angular
+  .module('App')
+  .controller('DedicatedCloudSecurityPolicyAddCtrl', class {
+    constructor(
```
Comment (JeremyDec) [inline client/app/dedicatedCloud/security/add/dedicatedCloud-security-add.controller.js]:
```suggestion
    /* @ngInject */
    constructor(
```

File: client/app/dedicatedCloud/security/session-timeout/update/dedicatedCloud-security-session-timeout-update.controller.js [hunk 8]
```diff
@@ -1,37 +1,48 @@
-class DedicatedCloudSecurityUpdateSessionTimeoutCtrl {
-  constructor($scope, $stateParams, DedicatedCloud, $translate) {
-    this.$scope = $scope;
-    this.$stateParams = $stateParams;
-    this.DedicatedCloud = DedicatedCloud;
-    this.$translate = $translate;
+{
+  class DedicatedCloudSecurityUpdateSessionTimeoutCtrl {
+    constructor(
```
Comment (JeremyDec) [inline client/app/dedicatedCloud/security/session-timeout/update/dedicatedCloud-security-session-timeout-update.controller.js]:
```suggestion
    /* @ngInject */
    constructor(
```

File: client/app/dedicatedCloud/security/simultaneous-connection/update/dedicatedCloud-security-max-simultaneous-connection.controller.js [hunk 9]
```diff
@@ -1,40 +1,50 @@
-class DedicatedCloudSecurityMaxSimultaneousConnectionsCtrl {
-  constructor($scope, $stateParams, DedicatedCloud, $translate) {
-    this.$scope = $scope;
-    this.$stateParams = $stateParams;
-    this.DedicatedCloud = DedicatedCloud;
-    this.$translate = $translate;
+{
+  class DedicatedCloudSecurityMaxSimultaneousConnectionsCtrl {
+    constructor(
```
Comment (JeremyDec) [inline client/app/dedicatedCloud/security/simultaneous-connection/update/dedicatedCloud-security-max-simultaneous-connection.controller.js]:
```suggestion
    /* @ngInject */
    constructor(
```

File: client/app/dedicatedCloud/user/delete/dedicatedCloud-user-delete.controller.js [hunk 10]
```diff
@@ -1,17 +1,39 @@
...
-        $scope.setMessage($translate.instant('dedicatedCloud_USER_delete_success', { t0: $scope.user.name }));
-      },
-      (data) => {
-        $scope.setMessage($translate.instant('dedicatedCloud_USER_delete_fail', { t0: $scope.user.name }), {
-          type: 'ERROR',
-          message: data.message,
+angular
+  .module('App')
+  .controller('DedicatedCloudUserDeleteCtrl', class {
+    constructor(
```
Comment (JeremyDec) [inline client/app/dedicatedCloud/user/delete/dedicatedCloud-user-delete.controller.js]:
```suggestion
    /* @ngInject */
    constructor(
```

File: client/app/dedicatedCloud/user/disable/dedicatedCloud-user-disable.controller.js [hunk 11]
```diff
@@ -1,14 +1,37 @@
...
-      (err) => {
-        $scope.setMessage($translate.instant('dedicatedCloud_USER_disable_fail', { t0: $scope.user.name }), err.data);
-      },
-    );
-  };
-});
+angular
+  .module('App')
+  .controller('DedicatedCloudUserDisableCtrl', class {
+    constructor(
```
Comment (JeremyDec) [inline client/app/dedicatedCloud/user/disable/dedicatedCloud-user-disable.controller.js]:
```suggestion
    /* @ngInject */
    constructor(
```

File: client/app/dedicatedCloud/user/enable/dedicatedCloud-user-enable.controller.js [hunk 12]
```diff
@@ -1,18 +1,34 @@
-angular.module('App').controller('DedicatedCloudUserEnableCtrl', ($scope, $stateParams, $translate, DedicatedCloud) => {
-  $scope.user = $scope.currentActionData;
+angular
+  .module('App')
+  .controller('DedicatedCloudUserEnableCtrl', class {
+    constructor(
```
Comment (JeremyDec) [inline client/app/dedicatedCloud/user/enable/dedicatedCloud-user-enable.controller.js]:
```suggestion
    /* @ngInject */
    constructor(
```

File: client/app/dedicatedCloud/user/enable/dedicatedCloud-user-enable.controller.js [hunk 13]
```diff
@@ -1,18 +1,34 @@
...
-      )
-      .finally(() => {
-        $scope.resetAction();
-      });
-  };
-});
+    $onInit() {
+      this.$scope.user = this.$scope.currentActionData;
+
+      this.enableUser = () => this.enableUser();
```
Comment (JeremyDec) [inline client/app/dedicatedCloud/user/enable/dedicatedCloud-user-enable.controller.js]:
```suggestion
      this.$scope.enableUser = () => this.enableUser();
```

File: client/app/dedicatedCloud/security/logout/dedicatedCloud-security-logout.controller.js [hunk 14]
```diff
@@ -1,36 +1,46 @@
-class DedicatedCloudSecurityPolicyLogoutCtrl {
-  constructor($scope, $stateParams, DedicatedCloud, $translate) {
-    this.$scope = $scope;
-    this.$stateParams = $stateParams;
-    this.DedicatedCloud = DedicatedCloud;
-    this.$translate = $translate;
+{
+  class DedicatedCloudSecurityPolicyLogoutCtrl {
+    constructor(
```
Comment (JeremyDec) [inline client/app/dedicatedCloud/security/logout/dedicatedCloud-security-logout.controller.js]:
```suggestion
    /* @ngInject */
    constructor(
```

**Pull request merged**